client/snap_op.go

```diff
@@ -305,3 +305,59 @@ func sendSnapFile(snapPath string, snapFile *os.File, pw *io.PipeWriter, mw *mul
 	mw.Close()
 	pw.Close()
 }
+
+type snapRevisionOptions struct {
+	Channel   string `json:"channel,omitempty"`
+	Revision  string `json:"revision,omitempty"`
+	CohortKey string `json:"cohort-key,omitempty"`
+}
+
+type downloadData struct {
```

Comment:
the `data` there doesn't add any information. Maybe `downloadAction` or `downloadInstruction`?

And yes, I know `snap_op` has `actionData`. That should clearly be `snapAction`. We've got a lot of things that have an Action in them: `appInstruction`, `snapshotAction`, `debugAction`, `aliasAction`, `CohortAction`, `InterfaceAction`, `warningsAction`, `actionData`, and`multiActionData`. The last two are clearly bad names: you have no idea what they are *for*. The rest could use some more uniformity, but no biggie.